File: WOT-iOS/WOTData/WOTData/core/requests/Guns/WOTWebRequestTankGuns.m
//
//  WOTWebRequestTankGuns.m
//  WOT-iOS
//
//  Created by Pavel Yeshchyk on 6/23/15.
//  Copyright (c) 2015 Pavel Yeshchyk. All rights reserved.
//

#import "WOTWebRequestTankGuns.h"
#import "WOTDataDefines.h"
//#import <WOTData/WOTData.h>

@implementation WOTWebRequestTankGuns

- (NSDictionary *)query {
    
    NSMutableDictionary *result = [[NSMutableDictionary alloc] init];
    result[WOTApiKeys.application_id] = [NSString valueOrSpaceString:self.hostConfiguration.applicationID];
    result[WOTApiKeys.fields] = [NSString valueOrSpaceString:[self.args escapedValueForKey:WOTApiKeys.fields]];
    result[WOTApiKeys.module_id] = [NSString valueOrSpaceString:[self.args escapedValueForKey:WOTApiKeys.module_id]];
    return result;
}

- (NSString *)path {
    
    return @"/wot/encyclopedia/tankguns/";
}

@end

File: WOT-iOS/WOTData/WOTData/WebServices/requests/WOTWEBRequestTankGuns.swift
//
//  WOTWEBRequestTankGuns.swift
//  WOTData
//
//  Created by Pavel Yeshchyk on 1/14/20.
//  Copyright © 2020 Pavel Yeshchyk. All rights reserved.
//

import Foundation

@objc
public class WOTWEBRequestTankGuns: WOTWEBRequest {
    override public var method: String { return "POST" }

    override public var path: String {
        return "/wot/encyclopedia/tankguns/"
    }
}

extension WOTWEBRequestTankGuns: WOTModelServiceProtocol {
    @objc
    public class func modelClass() -> AnyClass? {
        return Tankguns.self
    }

    @objc
    public func instanceModelClass() -> AnyClass? {
        return type(of: self).modelClass()
    }
}
